cmd/tealdbg/debugger.go

```diff
@@ -364,28 +364,26 @@ func (s *session) GetSourceMap() ([]byte, error) {
 	lines := make([]string, len(s.lines))
 	const targetCol int = 0
 	const sourceIdx int = 0
-	sourceLine := 0
-	const sourceCol int = 0
 	prevSourceLine := 0
 
 	// the very first entry is needed by CDT
-	lines[0] = logic.MakeSourceMapLine(targetCol, sourceIdx, 0, sourceCol)
+	lines[0] = logic.MakeSourceMapLine(targetCol, sourceIdx, 0, 0)
 	for targetLine := 1; targetLine < len(s.lines); targetLine++ {
 		if pc, ok := s.pcOffset[targetLine]; ok && pc != 0 {
-			sourceLine, ok = s.offsetToLine[pc]
+			source, ok := s.offsetToSource[pc]
 			if !ok {
 				lines[targetLine] = ""
 			} else {
-				lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, sourceLine-prevSourceLine, sourceCol)
-				prevSourceLine = sourceLine
+				lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, source.Line-prevSourceLine, source.Column)
```

Comment:
Similar to lines, we must report column deltas, while this code reports the absolute column index.

Suggestion: replace `prevSourceLine` with `prevLocation` (a `logic.SourceLocation`) so you can do `source.Column-prevLocation.Column`

Reply:
I didn't realize they were deltas. I agree on the suggestion.